File: SDKs/amazonaws.com/securityhub/2018-10-26/python/src/sdk/models/shared/insightresults.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import insightresultvalue as shared_insightresultvalue


@dataclass_json
@dataclasses.dataclass
class InsightResults:
    r"""InsightResults
    The insight results returned by the <code>GetInsightResults</code> operation.
    """
    
    group_by_attribute: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('GroupByAttribute') }})
    insight_arn: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('InsightArn') }})
    result_values: list[shared_insightresultvalue.InsightResultValue] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ResultValues') }})
    

File: SDKs/amazonaws.com/securityhub/2018-10-26/java/lib/src/main/java/openapisdk/models/shared/InsightResults.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * InsightResults
 * The insight results returned by the <code>GetInsightResults</code> operation.
**/
public class InsightResults {
    @JsonProperty("GroupByAttribute")
    public String groupByAttribute;
    public InsightResults withGroupByAttribute(String groupByAttribute) {
        this.groupByAttribute = groupByAttribute;
        return this;
    }
    @JsonProperty("InsightArn")
    public String insightArn;
    public InsightResults withInsightArn(String insightArn) {
        this.insightArn = insightArn;
        return this;
    }
    @JsonProperty("ResultValues")
    public InsightResultValue[] resultValues;
    public InsightResults withResultValues(InsightResultValue[] resultValues) {
        this.resultValues = resultValues;
        return this;
    }
}